File: ch1/Exercise_1.3.4.1/task05.cpp
#include <bits/stdc++.h>                         // C++ code for task 5
using namespace std;
// utilize the natural sort order of primitive data types in the tuple
typedef tuple<int, int, int> iii;
int main() {
  vector<iii> birthdays;
  birthdays.emplace_back(5, 24, -1980);          // reorder DD/MM/YYYY
  birthdays.emplace_back(5, 24, -1982);          // to MM, DD, and then
  birthdays.emplace_back(11, 13, -1983);         // use NEGATIVE YYYY
  sort(birthdays.begin(), birthdays.end());      // that's all :)
  for (auto &[mm, dd, yyyy] : birthdays)         // C++17 style
    printf("%d %d %d\n", dd, mm, -yyyy);
}

File: ch1/Exercise_1.3.4.1/task05.py
a = (5, 24, -1982)                                # Python code for task 5
b = (5, 24, -1980)
c = (11, 13, -1983)
s = sorted({c, a, b})
for i in s:
    print(f'{i[1]} {i[0]} {-i[2]}')
